### PR TITLE
MAINT: copy, array-api compatible utility function

### PR DESCRIPTION
**Pull request by andyfaff**
It seems to me that having an array-api compatible `copy` function would be quite useful. I have attempted to create one in this PR. Sorry, probably should've sent an item to scipy-dev, but I didn't think it would be controversial.

### CODE REVIEW

**Comment by lucascolley** [comment]
Agreed that this seems useful, https://github.com/scipy/scipy/pull/19005 would make use of this twice but there will probably be other submodules which would use it more.

File: scipy/_lib/_array_api.py [hunk 1]
```diff
@@ -136,3 +136,26 @@ def atleast_nd(x, *, ndim, xp):
         x = xp.expand_dims(x, axis=0)
         x = atleast_nd(x, ndim=ndim, xp=xp)
     return x
+
+
+def copy(x):
```
Comment (mdhaber) [inline scipy/_lib/_array_api.py:141]: Typically `copy` would not be the only operation that depends on the underlying array library. Do we want every utility function like this separately calling `array_namespace` to detect the namespace, or should `xp` be passed in?

**Review by mdhaber — COMMENTED** [review]
This would be private, so I don't think an email is necessary. One comment.

File: scipy/_lib/_array_api.py (hunk 1 above)
Comment (andyfaff) [reply scipy/_lib/_array_api.py:141]: Yes, we probably should pass it in to save effort. I think we should pass it in as a keyword:

```
def copy(x, *, xp=None):
    if xp is None:
        xp = array_namespace(x)
...
```

File: scipy/_lib/_array_api.py [hunk 2]
```diff
@@ -136,3 +136,30 @@ def atleast_nd(x, *, ndim, xp):
...
+
+def copy(x, *, xp=None):
+    """
+    Copies an array.
+
+    Parameters
+    ----------
+    x : array
+
+    x: array_namespace
```
Comment (mdhaber) [inline scipy/_lib/_array_api.py:149]:
```suggestion
    xp : array_namespace
```

**Review by mdhaber — APPROVED** [review]
It's arguable that we don't need a one-line wrapper of `as_xparray`, but I think it makes the intent ("copy this thing", not "make this an array") more clear, so it seems fine.

I checked out the other `array_api_compatible` tests and this looks consistent with those, so presumably it is actually being tested with a non-regular-NumPy backend (maybe `numpy.array_api`) on CI.

**Comment by andyfaff** [comment]
> I checked out the other array_api_compatible tests and this looks consistent with those, so presumably it is actually being tested with a non-regular-NumPy backend (maybe numpy.array_api) on CI.

That is correct. You can see from these failed [CI tests](https://github.com/scipy/scipy/actions/runs/5779075732/job/15660954322#step:10:67) that various backends are used. The CI [configuration](https://github.com/scipy/scipy/blob/main/.github/workflows/array_api.yml#L88) indicates that pytorch, numpy, numpy.array_api are examined.

**Pull request merged**